Block/AbstractTemplate.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * Snowdog
+ *
+ * @author      Dawid Czaja <dawid.czaja@snow.dog>.
+ * @category
+ * @package
+ * @copyright   Copyright Snowdog (http://snow.dog)
+ */
+
+namespace Snowdog\Menu\Block;
+
+use Magento\Framework\View\Element\Template;
+
+/**
+ * Class AbstractTemplate
+ * @package Snowdog\Menu\Block
+ */
+abstract class AbstractTemplate extends Template
```

Comment:
To avoid excessive inheritance IMO it is better to create a template resolver class. Next, inject the resolver to `\Snowdog\Menu\Block\NodeType\AbstractNode` and `\Snowdog\Menu\Block\Menu` to set the block template like:
```php
parent::__construct($context, $data);
$this->setTemplate($this->templateResolver->getTemplate($this));
```

Reply:
Now templateResolver is created and inject to abstractNode and Menu please check